api/events.go

```diff
@@ -80,29 +80,31 @@ func (eh *eventsHandler) cleanupHandler(remoteAddr string) {
 
 }
 
-// Handle writes information about a cluster event to each remote address in the cluster that has been added to the events handler.
-// After an unsuccessful write to a remote address, the associated channel is closed and the address is removed from the events handler.
-func (eh *eventsHandler) Handle(e *cluster.Event) error {
+// normalizeEvent takes a cluster Event and ensures backward compatibility
+// and all the right fields filled up
+func normalizeEvent(receivedEvent *cluster.Event) ([]byte, error) {
+	// make a local copy of the event
+	e := *receivedEvent
```

Comment:
While it is slightly wasteful if you're not writing to it, it might be worthwhile to just copy the whole structure here to avoid future bugs. These can be nasty in practice.